public/scripts/scripts.js

```diff
@@ -1,3 +1,5 @@
+var activeTrack;
```

Comment:
fwiw a pattern that i've found useful in tracking global variables is to collect them together into a single object, like

```
var state = {activeTrack: null};
```

and then later in code, you can set `state.activeTrack = foo`, and you can read from `state.activeTrack`, etc. this obviously doesn't make a huge difference right now if you've only got one global variable, but i find that collecting them all together in a named object has some nice effects:

- it makes it clear when you're referring to one of these accursed global variables vs when you're referring to a benign local variable [this is the primary reason that this pattern is useful imo]
- seeing your global variables defined all together next to each other makes it really obvious how many there are, so if you end up making too many of them you'll know sooner than later, because you'll see them all right there and it'll feel icky
- having them all defined in one place gives you an opportunity to document them all - eg for each one you can say what sorts of values it can take on, what reads it, what modifies it, what its general semantics are, etc (although you could also totally do this if they were defined as separate variables)

like we talked about, it's fine to have a very small number of global mutable variables, as long as they're each only modified in one or maybe two places, and as long as it's not possible for another part of the program to observe the variable in a half-formed state (e.g. a half-full allCallSongs, etc). simple primitive values like strings and integers are fine. if you end up with more than ~3-4 global variables or if any of them start having complicated values that are written to in many places or that can be seen half-formed, that's where you can run into issues. glgl!